File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/IndexGroup.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description:
 */
@Target(ElementType.ANNOTATION_TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface IndexGroup {

    int GENERIC = -1;

    /**
     * @return The number that each contained {@link Index} points to, so they can be combined into a single index.
     * If {@link #GENERIC}, this will assume a generic index that covers the whole table.
     */
    int number() default GENERIC;

    /**
     * @return The name of this index. It must be unique from other {@link IndexGroup}.
     */
    String name();

    /**
     * @return If true, this will disallow duplicate values to be inserted into the table.
     */
    boolean unique() default false;
}

File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/IndexGroup.kt
package com.raizlabs.android.dbflow.annotation

const val INDEX_GENERIC = -1

/**
 * Description:
 */
@Target(AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class IndexGroup(
        /**
         * @return The number that each contained [Index] points to, so they can be combined into a single index.
         * If [.INDEX_GENERIC], this will assume a generic index that covers the whole table.
         */
        val number: Int = INDEX_GENERIC,
        /**
         * @return The name of this index. It must be unique from other [IndexGroup].
         */
        val name: String,
        /**
         * @return If true, this will disallow duplicate values to be inserted into the table.
         */
        val unique: Boolean = false)

